src/g3log/g3log.hpp

```diff
@@ -26,7 +26,6 @@
 #include "g3log/logmessage.hpp"
 
 #include <string>
-#include <cstdarg>
```

Comment:
please put back. It's included elsewhere which I am aware of but I am of the firm opinion that each header file should include what is needed in that file. 

I.e.  <cstdarg> is needed for the following
```
 #define CHECKF(boolean_expression, printf_like_message, ...)    \
    if (false == (boolean_expression))  INTERNAL_CONTRACT_MESSAGE(#boolean_expression).capturef(printf_like_message, ##__VA_ARGS__)
 ```

Please take a look at the other includes you removed in the other files too.   I appreciate that you took the time and effort to prune them. Please put back the once that are needed in the definitions file (.hpp)